pennylane/pauli/pauli_arithmetic.py

```diff
@@ -0,0 +1,280 @@
+        """Given a Pauli word, get the matrix representation.
+
+        KeywordArgs:
+            wire_order (iterable or None): The order of qubits in the tensor product.
+            format (str): The format of the matrix ("dense" by default), if not a dense
+                matrix, then the format for the sparse representation of the matrix.
+
+        Returns:
+            (Union[NumpyArray, ScipySparseArray]): Matrix representation of the Pauliword
+
+        Raises:
+            ValueError: Can't get the matrix of an empty PauliWord.
+        """
+        if len(self) == 0:
+            if wire_order is None or wire_order == wires.Wires([]):
+                raise ValueError("Can't get the matrix of an empty PauliWord.")
+            return (
+                np.eye(2 ** len(wire_order))
+                if format == "dense"
+                else sparse.eye(2 ** len(wire_order), format=format)
+            )
+
+        matrix_map = sparse_mat_map if format != "dense" else mat_map
+        kron = sparse.kron if format != "dense" else math.kron
+
+        return reduce(kron, (matrix_map[self[w]] for w in wire_order))
+
+
+class PauliSentence(dict):
+    """Dict representing a Pauli Sentence. The keys are
+    PauliWord instances and the values correspond to coefficients.
+
+    >>> ps = PauliSentence({
+            PauliWord({0:X, 1:Y}): 1.23
+            PauliWord({2:Z, 0:Y}): -0.45j
+        })
+    """
+
+    def __missing__(self, key):
+        """If the PauliWord is not in the sentence then the coefficient
+        associated with it should be 0."""
+        return 0.0
+
+    def __add__(self, other):
+        """Add two Pauli sentence together by iterating over the smaller
+        one and adding its terms to the larger one."""
+        smaller_ps, larger_ps = (
+            (self, copy(other)) if len(self) < len(other) else (other, copy(self))
+        )
+        for key in smaller_ps:
+            larger_ps[key] += smaller_ps[key]
+
+        return larger_ps
+
+    def __mul__(self, other):
+        """Multiply two Pauli sentences by iterating over each sentence and multiplying
+        the Pauli words pair-wise"""
+        final_ps = PauliSentence()
+
+        for pw1 in self:
+            for pw2 in other:
+                prod_pw, coeff = pw1 * pw2
+                final_ps[prod_pw] += coeff * self[pw1] * other[pw2]
+
+        return final_ps
+
+    def __str__(self):
+        """String representation of the PauliSentence."""
+        return "\n+ ".join(f"{coeff} * {str(pw)}" for pw, coeff in self.items())
+
+    def __repr__(self):
+        """Terminal representation for PauliSentence"""
+        return str(self)
+
+    @property
+    def wires(self):
+        """Track wires of the PauliSentence."""
+        return set().union(*(pw.wires for pw in self.keys()))
+
+    def to_mat(self, wire_order, format="dense"):
+        """Given a Pauli word, get the matrix representation.
+
+        KeywordArgs:
```

Comment:
```suggestion
        Keyword Args:
```
Some kind of rendering issue in the documentation. Maybe this will fix it?